pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConfig.java

```diff
@@ -172,7 +172,7 @@ public StreamConfig(String tableNameWithType, Map<String, String> streamConfigMa
 
     _flushThresholdRows = extractFlushThresholdRows(streamConfigMap);
     _flushThresholdTimeMillis = extractFlushThresholdTimeMillis(streamConfigMap);
-    _flushThresholdSegmentSizeBytes = extractFlushThresholdSegmentSize(streamConfigMap);
+    _flushThresholdSegmentSizeBytes = extractFlushThresholdSegmentSize(streamConfigMap, _flushThresholdRows);
```

Comment:
Do not add check within the constructor because it will break existing table configs. For new checks we want to enforce, add it into `TableConfigUtils.validate()`